src/compile/common.ts

```diff
@@ -99,10 +99,19 @@ export function formatSignalRef(fieldDef: FieldDef<string>, specifiedFormat: str
   } else if (fieldDef.type === 'temporal') {
     const isUTCScale = isScaleFieldDef(fieldDef) && fieldDef['scale'] && fieldDef['scale'].type === ScaleType.UTC;
     return {
-      signal: timeFormatExpression(field(fieldDef, {expr}), fieldDef.timeUnit, specifiedFormat, config.text.shortTimeLabels, config.timeFormat, isUTCScale)
+      signal: timeFormatExpression(field(fieldDef, {expr}), fieldDef.timeUnit,
+        specifiedFormat, config.text.shortTimeLabels, config.timeFormat, isUTCScale)
     };
   } else {
-    return {signal: field(fieldDef, {expr})};
+    if (!specifiedFormat || specifiedFormat === 'number') {
```

Comment:
The predicate logic here seems pretty weird.    From a quick skim, there are multiple problems:

- `format` and `formatType` are two different things. 
- There are still many cases where you shouldn't add `format` (e.g., normal string)
- Why is else case adding timeFormatExpression

Revise your logic!